packages/toolkit/src/query/react/buildHooks.ts

```diff
@@ -827,7 +827,8 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
       const reset = useCallback(() => {
         if (promise) {
           setPromise(undefined)
-        } else if (fixedCacheKey) {
+        }
+        if (fixedCacheKey) {
           dispatch(
             api.internalActions.removeMutationResult({
               requestId,
```

Comment:
Hmm. Do we want to wrap the whole thing in a `batch` now? :thinking:

Reply:
In it's current state without `batch`, calling `reset()` only seems to result in one render for each component using the hook, including the one that called the original `trigger`. Is there some other benefit that `batch` will provide?

Regardless of the above, I'm not against batching it anyway as a "might help, and can't hurt" scenario

Reply:
It depends how you call it. If you call it in an effect or click handler, React will automatically batch and only rerender once. If you do it in something async, for example after a timeout, it would lead to two renders

Reply:
ah, for some reason I assumed click handlers were exempt from the default batching. It's definitely worth batching this, I'll change that